lib/test/mdx_test.ml

```diff
@@ -321,6 +321,7 @@ let run_exn ~non_deterministic ~silent_eval ~record_backtrace ~syntax ~silent
             Toplevel.of_lines ~syntax ~loc:t.loc t.contents
           in
           Deprecated.Missing_double_semicolon.check_block phrases;
+          let phrases = Deprecated.Missing_double_semicolon.fix phrases in
```

Comment:
This will run through the commands twice for each of them, how about have the `fix` function also perform the check, renaming it on the way to make it clear it does both eventually?

Reply:
Ah yes, I somehow didn't see the forest for the trees. This is much nicer indeed and avoids nearly identical code.